libretranslate/docker-compose.yml

```diff
@@ -0,0 +1,36 @@
+version: "3"
+
+services:
+  app_proxy:
+    environment:
+      # <app-id>_<web-container-name>_1
+      # e.g. 'btc-rpc-explorer_web_1'
+      # Note that the '_1' at the end is needed
+      APP_HOST: libretranslate_main_1
+      APP_PORT: 5000
+      PROXY_AUTH_ADD: "false"
+  main:
+    image: libretranslate/libretranslate:v1.5.6
+    restart: unless-stopped
+
+    ## Uncomment this for logging in docker compose logs
+    # tty: true
+    healthcheck:
+      test: ['CMD-SHELL', './venv/bin/python scripts/healthcheck.py']     
+    ## Uncomment above command and define your args if necessary
+    # command: --ssl --ga-id MY-GA-ID --req-limit 100 --char-limit 500
+    ## Uncomment this section and the libretranslate_api_keys volume if you want to backup your API keys
+    environment:
+      - LT_API_KEYS=true
+      - LT_API_KEYS_DB_PATH=/app/db/api_keys.db # Same result as `db/api_keys.db` or `./db/api_keys.db`
+    ## Uncomment these vars and libretranslate_models volume to optimize loading time.
+      - LT_UPDATE_MODELS=true
+    ##  - LT_LOAD_ONLY=en,fr
+    volumes:
+      - libretranslate_api_keys:/app/db
+      ## Using $APP_DATA_DIR causes permission errors
+      - libretranslate_models:/home/libretranslate/.local:rw
+
+volumes:
+  libretranslate_models:
+  libretranslate_api_keys:
```

Comment:
> \## Using $APP_DATA_DIR causes permission errors

Ah, ya permissions issues can be super tricky in some of these apps. We'll need to figure out a workaround, and then change this from named volumes to bind mounts like we do for other apps so that we can have data directories in the same place for every user (i.e., `/home/umbrel/umbrel/app-data/<specific-app>`).

What's likely happening here is that the Dockerfile for libretranslate is defining a particular user that the container is running as, and this user won't have permissions to read/write to a bind mount owned by the umbrel user. We may be able to get around this just by passing the service `user: "1000:1000"`. I'll take a look at the Dockerfile and let you know.

Reply:
Hmmm, okay I just took a look at the Dockerfile: https://github.com/LibreTranslate/LibreTranslate/blob/f7a35db05b3ac7fcd40489faa7da827b03354faf/docker/Dockerfile#L28-L33

They do indeed define a user (`libretranslate`), but we're actually locked out from using `user: "1000:1000"` in the compose file to fix the permissions issue because the Dockerfile also specifically chowns the /app data directory:

```
USER libretranslate


COPY --from=builder --chown=1032:1032 /app /app
WORKDIR /app


COPY --from=builder --chown=1032:1032 /app/venv/bin/ltmanage /usr/bin/
```

I haven't tested this, but I think what will work is to just use a pre-start hook that runs before the container is spun up. The pre-start hook can chown the bind mounts to `1032:1032`.

Reply:
This will probably be easiest for me to do this. I'll push directly to your branch with atomic commits so you can see what I've done.